sql/catalyst/src/main/scala/org/apache/spark/sql/types/Decimal.scala

```diff
@@ -69,6 +68,16 @@ final class Decimal extends Ordered[Decimal] with Serializable {
   }
 
   /**
+   * Set this Decimal to the given BigInt. Will have precision 38 and scale 0.
+   */
+  def set(intVal: BigInt): Decimal = {
+    this.decimalVal = null
+    this.longVal = intVal.toLong
```

Comment:
Hi Davies: Yes, we need to check the range, otherwise, it will cause overflow. Thanks. I will look into.